go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -137,12 +137,24 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	}
 
 	// Set pod limits.
-	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" && sparkConfig["spark.driver.cores"] != "" {
-		sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.driver.cores"]
+	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" {
+		// spark.kubernetes.driver.request.cores takes precedence over spark.driver.cores
+		if sparkConfig["spark.kubernetes.driver.request.cores"] != "" {
+			sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.kubernetes.driver.request.cores"]
+		} else if sparkConfig["spark.driver.cores"] != "" {
+			sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.driver.cores"]
```

Comment:
Should you set the requests to match this same number?

Reply:
Don't think that's necessary as It's ok to have either `spark.kubernetes.driver.request.cores` or `spark.driver.cores` in the conf i.e. one is good enough. 
If both are present, `spark.kubernetes.driver.request.cores` has precedence over `spark.driver.cores`.

Reply:
SG